core/src/executor/evaluate/function.rs

```diff
@@ -274,6 +278,23 @@ pub fn floor(name: String, n: Evaluated<'_>) -> Result<Value> {
     Ok(Value::F64(eval_to_float!(name, n).floor()))
 }
 
+pub fn rand(seed: Option<Evaluated<'_>>) -> Result<Value> {
+    let v = match seed {
+        Some(seed) => seed.try_into()?,
+        None => {
+            return Ok(Value::F64(rand::random()));
+        }
+    };  
+        
+    let v = match v {
+        Value::I8(v) => StdRng::seed_from_u64(v as u64).gen(),
+        Value::I64(v) => StdRng::seed_from_u64(v as u64).gen(),
+        _ => rand::random(), // may be we need to return error?
```

Comment:
If you branch here, it seems to be an error because the value of 'seed' is of a type other than integer + signed integer.